### DISCUSSION TITLE
Add support for @container queries :)

### DISCUSSION THREAD

**Discussion by kolber** [vanilla-extract-css/vanilla-extract#624] [Ideas]
This is coming to browsers soon (currently available behind a flag on Chrome Canary), and already has a very solid polyfill available, and is of course insanely useful. :)

https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Container_Queries

https://github.com/GoogleChromeLabs/container-query-polyfill

It can work in the exact same way as @media currently does, so only requires very minimal changes to the codebase.

I got very basic support working by cloning `transformMedia()`, here: https://github.com/kolber/vanilla-extract/commit/94930dcc00623a63d8440797b80491c40608bec9

But I imagine `transformMedia()` could be rewritten to handle both syntaxes in a nice clean way maybe?

**Comment by mattcompiles** [comment]
Opened a PR to add support for container queries
https://github.com/seek-oss/vanilla-extract/pull/807

**Comment by tuzmusic** [comment]
I was able to use `createContainer` but I couldn't get it to actually work. The following works if you don't use the container name at all (don't assign it and don't refer to it inside `wideStyle`) but not either of the following ways:

```ts
const containerName = createContainer()

export const Card = style([
  {
    containerType: 'inline-size',
    containerName,
    minWidth: rem(288), // size of a large card at width = 320px
  },
])

const wideStyle = (style: StyleRule) => ({
  '@container': {
    [`${containerName} (min-width: 321px)`]: style,
  },
})

/* this doesn't work either
const wideStyle = (style: StyleRule) => ({
  [`@container ${containerName} `]: {
    [`(min-width: 321px)`]: style,
  },
})
*/

export const CardContent = style([
  treats({
    display: 'flex',
    flexDirection: 'column',
  }),
  wideStyle({
    flexDirection: 'row',
  }),
])
```

**Reply by askoufis** [reply]
@tuzmusic If you think you've found a bug, please raise an issue with a minimal reproduction.
